rzup/src/lib.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod cli;
+pub mod errors;
+pub mod extension;
+pub mod repo;
+pub mod toolchain;
+pub mod utils;
+
+use std::fs;
+
+use anyhow::{anyhow, Context, Result};
+
+pub struct Rzup;
+
+impl Rzup {
+    pub fn version() -> &'static str {
+        env!("CARGO_PKG_VERSION")
+    }
+    pub async fn update() -> Result<()> {
+        // TODO
```

Comment:
Updating self (rzup) is an important feature but with s3 hosting, gathering metadata about builds will differ from the toolchain or extension update process. Perhaps this can be skipped entirely and just run the init script?